### DISCUSSION TITLE
How to break sentence like "hello, can you hear me?" to 2 segments by actual timestamp?

### DISCUSSION THREAD

**Discussion by neoyxm** [SYSTRAN/faster-whisper#209] [Q&A]
Hi , I have an audio like below: between **hello** and **can you hear me**, there is about 5s delay. 
<img width="1042" alt="截屏2023-05-08 15 28 59" src="https://user-images.githubusercontent.com/1170080/236763252-d9050884-4993-4a36-8d89-034d73c33e50.png">

What I expect is  '**hello**' and '**can you hear me**' are put to 2 segments:
`    {
        "start": "00:01.680",
        "end": "00:02.448",
        "txt": "Hello"
    }`
`    {
        "start": "00:06.256",
        "end": "00:07.360",
        "txt": "can you hear me"
    }`

But they are put in one segment:
`    {
        "start": "00:00:06.256",
        "end": "00:00:07.360",
        "txt": "Hello, can you hear me?"
    }`

**Hello** is merged to  **can you hear me**  segment, and segment.start (**00:00:06.256**) is not  **Hello**'s start time (**00:01.680**).

My transcribe option is :  language=en, word_timestamps=True, vad_filter=True and others are default.

From the debug info from transcribing, the Silero VAD has splitted these two segments, but the result still puts them together.
`[INFO]-(transcribe:240) Processing audio with duration 00:09.280`
`[INFO]-(transcribe:249) VAD filter removed 00:04.944 of audio`
`[DEBUG]-(transcribe:255) VAD filter kept the following audio segments: [00:01.136 -> 00:02.448], [00:06.256 -> 00:09.280]
`

Is there any options which can help to put them to 2 segments by actual timestamp?

**Comment by neoyxm** [comment]
After set  word_timestamps = False,  the segment.start is ok

**Comment by neoyxm** [comment]
reopen it.

**Comment by palladium123** [comment]
If I understand correctly, you want to transcribe each speech chunks separately? 

If that is the case, your best bet is to split the audio separately by calling `get_speech_timestamps` on the audio file. This should return a List of start and end timestamps for each speech chunks, removing non-speech parts. You then want to call `collect_chunks` on each chunks returned from `get_speech_timestamps`. These two methods are in the `vad.py` module. Finally, call `transcribe` on each chunks with `vad_filter=False` (since you already used Silero VAD in the previous 2 steps). 

Haven't tested the above approach - it's just my take looking at the `transcribe` method.

**Reply by neoyxm** [reply]
Thanks, I will try that.

**Reply by zacharynapier** [reply]
Did you ever find a solution?

**Reply by neoyxm** [reply]
> Did you ever find a solution?


not yet

**Reply by toanhn** [reply]
I tried this approach, but sometimes I get more incorrect words or sentences in the resulting transcript than without using get_speech_timestamps.

**Comment by toanhuynhnguyen** [comment]
You can use Batched Faster-Whisper since it can transcribe an audio file to a transcript with word timestamps more accurate than sequence Faster-Whipser. Then you can separate a sentence into multiple sentences based on paused durations between words, such as 2 seconds or any N seconds you want.
